Sources/SwiftyStoreKit/PaymentsController.swift

```diff
@@ -64,12 +64,18 @@ public struct PaymentDiscount {
 class PaymentsController: TransactionController {
 
     private var payments: [Payment] = []
+    private var failedPayments: [Payment] = []
 
     private func findPaymentIndex(withProductIdentifier identifier: String) -> Int? {
-        for payment in payments where payment.product.productIdentifier == identifier {
-            return payments.firstIndex(of: payment)
-        }
-        return nil
+        return payments.enumerated()
+            .first { $0.element.product.productIdentifier == identifier }
+            .map { $0.offset }
+    }
+
+    private func findFailedPaymentIndex(withProductIdentifier identifier: String) -> Int? {
+        return failedPayments.enumerated()
+            .first { $0.element.product.productIdentifier == identifier }
+            .map { $0.offset }
```

Comment:
I suggest to refactor this duplicate code

Reply:
Could just use `firstIndex` instead of enumerated + first + map here I guess. I think a private array extension with Payment as Element would do, something like

```swift
private extension Array where Element == Payment {
  func firstIndex(withIdentifier identifier: String) -> Element? {
     return firstIndex { $0.product.productIdentifier == identifier }
  }
}
```

then all find***Index functions could be removed and this extension could be used in code as in  `failedPayments.firstIndex(withIdentifier: <identifier>)`

Reply:
minor corrections to my example extension code:
1. Int? result, not Element?
2. I'd call external arg label withProductIdentifier to better match with its usage & keep it in line with labels in current find funcs

Reply:
Good idea! Updated to something similar to what @vapor-pawelw suggested in c9db0e6